src/otx/hpo/resource_manager.py

```diff
@@ -161,49 +149,96 @@ def release_resource(self, trial_id: Any):
         if trial_id not in self._usage_status:
             logger.warning(f"{trial_id} trial don't use resource now.")
         else:
-            self._available_gpu.extend(self._usage_status[trial_id])
+            self._available_devices.extend(self._usage_status[trial_id])
             del self._usage_status[trial_id]
 
     def have_available_resource(self):
         """Check that there is available resource."""
-        return len(self._available_gpu) >= self._num_gpu_for_single_trial
+        return len(self._available_devices) >= self._num_devices_per_trial
+
+
+class GPUResourceManager(AcceleratorManager):
+    """Resource manager class for GPU."""
+
+    def _set_available_devices(self, available_devices: Optional[str] = None) -> List[int]:
+        if available_devices is None:
+            cuda_visible_devices = os.getenv("CUDA_VISIBLE_DEVICES")
+            if cuda_visible_devices is not None:
+                available_devices_arr = _cvt_comma_delimited_str_to_list(cuda_visible_devices)
+            else:
+                num_gpus = torch.cuda.device_count()
+                available_devices_arr = list(range(num_gpus))
+        else:
+            available_devices_arr = _cvt_comma_delimited_str_to_list(available_devices)
+
+        return available_devices_arr
+
+    def _make_env_var_for_train(self, device_arr: List[int]) -> Dict[str, str]:
+        return {"CUDA_VISIBLE_DEVICES": ",".join([str(val) for val in device_arr])}
+
+
+class XPUResourceManager(AcceleratorManager):
+    """Resource manager class for XPU."""
+
+    def _set_available_devices(self, available_devices: Optional[str] = None) -> List[int]:
+        if available_devices is None:
+            cuda_visible_devices = os.getenv("ONEAPI_DEVICE_SELECTOR", "").split(":")
```

Comment:
Why cuda is mentioned here?

Reply:
Changed at https://github.com/openvinotoolkit/training_extensions/pull/2704/commits/2607ab104c63df4712506cc667c48a305a60ffbc